ignite/distributed/auto.py

```diff
@@ -63,12 +63,11 @@ def auto_dataloader(dataset, **kwargs):
     world_size = idist.get_world_size()
 
     logger = setup_logger(__name__ + ".auto_dataloader")
-
     if world_size > 1:
-        if "batch_size" in kwargs:
+        if "batch_size" in kwargs and kwargs["batch_size"] >= world_size:
             kwargs["batch_size"] //= world_size
 
-        if "num_workers" in kwargs:
+        if "num_workers" in kwargs and kwargs["num_workers"] >= world_size:
```

Comment:
Actually, just realized that here we should check if `num_workers >= dist.get_nproc_per_node()` and not world size...

Reply:
I have removed it because I have no arguments for keeping it